lib/p2p/Pool.ts

```diff
@@ -36,6 +37,11 @@ type PoolConfig = {
   addresses: string[];
 };
 
+type NodeInfo = {
```

Comment:
I think we should either rename this to `NodeReputationInfo`, or just drop this type and let the return value for `getNodeReputation` be implied, since we're not using the type anywhere else.